test/parallel/test-fs-open.js

```diff
@@ -32,8 +32,15 @@ fs.open(__filename, 'rs', function(err, fd) {
   openFd2 = fd;
 });
 
+assert.throws(function() {
+  fs.open('/path/to/file/that/does/not/exist', { bad: 'flags' }, assert.fail);
+}, function(err) {
+  if (err instanceof TypeError) {
+    return err.message === 'File open flags may only be string or integer';
+  }
+});
```

Comment:
It seems that I cannot do what you suggest *and* what @cjihrig suggests in one go.

The [docs](https://nodejs.org/docs/latest/api/assert.html#assert_assert_throws_block_error_message) are really ambiguous about which argument does what (apart from the first argument). However neither of these work:

```js
assert.throws(function () { throw new TypeError('Hello'); }, /Hello/, RangeError);
assert.throws(function () { throw new TypeError('Hello'); }, TypeError, 'Bye');
```

Both asserts pass. So while a regexp can be passed, and a constructor function can be passed, it seems they can't both be passed.